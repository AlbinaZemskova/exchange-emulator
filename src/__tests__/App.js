import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson } from 'enzyme-to-json';
import {
  App,
  mapDispatchToProps,
} from '../App';
import 'jest-prop-type-error';

Enzyme.configure({ adapter: new Adapter() });

describe('<App />', () => {
  let wrapper;
  let addOrderForBuy;
  let addOrderForSell;
  let loadOrdersForBuy;
  let loadOrdersForSell;
  let loadHistoryOfDeal;

  beforeEach(() => {
    addOrderForBuy = jest.fn();
    addOrderForSell = jest.fn();
    loadOrdersForSell = jest.fn();
    loadHistoryOfDeal = jest.fn();
    loadOrdersForBuy = jest.fn();

    wrapper = shallow(
      <App
        addOrderForBuy={addOrderForBuy}
        addOrderForSell={addOrderForSell}
        loadOrdersForBuy={loadOrdersForBuy}
        loadOrdersForSell={loadOrdersForSell}
        loadHistoryOfDeal={loadHistoryOfDeal}
      />,
    );
  });

  describe('App component', () => {
    describe('render', () => {
      it('renders without crashing, generate orders from config for first launch', () => {
        expect(addOrderForBuy).toHaveBeenCalledTimes(15);
        expect(addOrderForSell).toHaveBeenCalledTimes(15);
        expect(JSON.parse(localStorage.getItem('serverStorage'))).toEqual({
          buyOrders: [],
          sellOrders: [],
          history: [],
        });

        expect(shallowToJson(wrapper)).toMatchSnapshot();
      });

      it('checking correct proptypes', () => {
        expect(() => (
          <App
            addOrderForBuy={{}}
            addOrderForSell={{}}
            loadOrdersForBuy={{}}
            loadOrdersForSell={{}}
            loadHistoryOfDeal={{}}
          />
        )).toThrow();
      });

      it('load orders', () => {
        expect(JSON.parse(localStorage.getItem('isInitialValuesWasSetted'))).toBe(true);
        expect(loadOrdersForBuy).toHaveBeenCalled();
        expect(loadOrdersForSell).toHaveBeenCalled();
        expect(loadHistoryOfDeal).toHaveBeenCalled();
      });
    });
  });

  describe('mapDispatchToProps', () => {
    const dispatch = jest.fn();

    const actions = mapDispatchToProps(dispatch);
    actions.loadOrdersForSell();
    expect(dispatch).toHaveBeenCalled();
  });
});
