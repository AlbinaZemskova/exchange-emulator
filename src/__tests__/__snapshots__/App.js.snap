// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> App component render renders without crashing, generate orders from config for first launch 1`] = `
<Fragment>
  <Connect(InputDataForOrder) />
  <Connect(RenderOrders) />
  <Connect(HistoryOfDeals) />
</Fragment>
`;
